zanata-war/src/main/java/org/zanata/util/HmacUtil.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2015, Red Hat, Inc. and individual contributors as indicated by the
+ * @author tags. See the copyright.txt file in the distribution for a full
+ * listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This software is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA, or see the FSF
+ * site: http://www.fsf.org.
+ */
+
+package org.zanata.util;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.codec.binary.StringUtils;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+
+/**
+ * @author Alex Eng <a href="mailto:aeng@redhat.com">aeng@redhat.com</a>
+ */
+public class HmacUtil {
+
+    private static final String HMAC_SHA1_ALGORITHM = "HmacSHA1";
+
+    private static final String HMAC_MD5_ALGORITHM = "HmacMD5";
+
+    private static final String HMAC_SHA256_ALGORITHM = "HmacSHA256";
+
+    public static String hmacSha1(final String key, final String valueToDigest)
+            throws IllegalArgumentException {
+        return hmacSha1(StringUtils.getBytesUtf8(key),
```

Comment:
Maybe add a javadoc about the UTF-8 encoding.  

It's important that the character encoding used here, and the character encoding used when encoding our JSONType objects, are matching.  It's extremely hard to verify that by reading code, with the current implementation.